src/ILCompiler.Compiler/src/Compiler/MetadataManager.cs

```diff
@@ -182,13 +182,18 @@ protected virtual void Graph_NewMarkedNode(DependencyNodeCore<NodeFactory> obj)
 
             if (methodNode != null)
             {
+                if (methodNode.Method.OwningType.GetTypeDefinition() is INonEmittableType && IsMethodEligibleForTracking(methodNode.Method))
```

Comment:
I'm a bit worried about this line.

Is it possible the need for this if block comes from the metadata policy needing tweaks? Do we still need it after addressing the other feedback?

Reply:
I tried to remove this one after addressing the other feedback and it hits the assert at https://github.com/dotnet/corert/blob/74897d4ab3eaff1141f8ac07287ccefbfbb5a352/src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs#L537 with stack
```
         Assertion Failed

            at ILCompiler.DependencyAnalysis.NodeFactory.NecessaryTypeSymbol(TypeDesc type) in D:\GitHub\corert\src\ILCompiler.Compiler\src\Compiler\DependencyAnalysis\NodeFactory.cs:line 538
            at ILCompiler.DependencyAnalysis.ExactMethodInstantiationsNode.GetData(NodeFactory factory, Boolean relocsOnly) in D:\GitHub\corert\src\ILCompiler.Compiler\src\Compiler\DependencyAnalysis\ExactMethodInstantiationsNode.cs:line 68
            at ILCompiler.DependencyAnalysis.WebAssemblyObjectWriter.EmitObject(String objectFilePath, IEnumerable`1 nodes, NodeFactory factory, WebAssemblyCodegenCompilation compilation, IObjectDumper dumper) in D:\GitHub\corert\src\ILCompiler.WebAssembly\src\CodeGen\WebAssemblyObjectWriter.cs:line 772
            at ILCompiler.WebAssemblyCodegenCompilation.CompileInternal(String outputFile, ObjectDumper dumper) in D:\GitHub\corert\src\ILCompiler.WebAssembly\src\Compiler\WebAssemblyCodegenCompilation.cs:line 54
            at ILCompiler.Compilation.ILCompiler.ICompilation.Compile(String outputFile, ObjectDumper dumper) in D:\GitHub\corert\src\ILCompiler.Compiler\src\Compiler\Compilation.cs:line 387
            at ILCompiler.Program.Run(String[] args) in D:\GitHub\corert\src\ILCompiler\src\Program.cs:line 576
            at ILCompiler.Program.Main(String[] args) in D:\GitHub\corert\src\ILCompiler\src\Program.cs:line 732
```